### PR TITLE
push up variable declaration

### PR DESCRIPTION
**Pull request by ericgribkoff**
Fix another mistake from #26338 (mainly applying to runs with `--use_existing_gcp_resources`) where the variable in the finally block may not have been declared

### CODE REVIEW

**Comment by ericgribkoff** [comment]
Ok, tested this with a number of manually triggered runs and I think I've finally got it right. Moved the sponge log report generation into the finally block (as-is, this was not being executed on calling sys.exit and logs for all test cases were lost, and we also need those to debug the failure).

File: tools/run_tests/run_xds_tests.py [hunk 1]
```diff
@@ -3274,18 +3270,18 @@ def __init__(self, compute, alpha_compute, project, project_num):
...
-        report_utils.render_junit_xml_report(test_results,
-                                             os.path.join(
-                                                 _TEST_LOG_BASE_DIR,
-                                                 _SPONGE_XML_NAME),
-                                             suite_name='xds_tests',
-                                             multi_target=True)
+
         if failed_tests:
             logger.error('Test case(s) %s failed', failed_tests)
             sys.exit(1)
```
Comment (lidizheng) [inline tools/run_tests/run_xds_tests.py:3276]: Do we want all failed runs to be saved? If GCP has an outage, our quota might run out.

Comment (ericgribkoff) [reply tools/run_tests/run_xds_tests.py:3276]: Maybe I'm missing something, but I think this only saves resources if `skip_cleanup_to_debug_failure` is set to true, which only happens when the load_report test fails. The changed line here just moves the call to `render_junit_xml_report` to the finally block.

Comment (ericgribkoff) [reply tools/run_tests/run_xds_tests.py:3276]: Also, for what it's worth, this change should only need to be live for ~1 day before we "catch" a failing instance of the load report test, and until then I'll just be manually babysitting the project (aka, running the cleanup script) to make sure we don't save too many sets of backends.

Comment (lidizheng) [reply tools/run_tests/run_xds_tests.py:3276]: TIL again that `sys.exit` raises `SystemExit`, it's not a hard halting at all. LGTM.

**Review by lidizheng — APPROVED** [review]

**Comment by lidizheng** [comment]
BTW, I saw multiple failed load-based-failover runs being preserved, e.g.: https://testgrid.corp.google.com/grpc-lb#gRPC%20Python%20xDS%20Master

**Pull request merged**